### ISSUE TITLE
[REVIEW]: ytree: A Python package for analyzing merger trees

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1881]
**Submitting author:** @brittonsmith (<a href="http://orcid.org/0000-0002-6804-630X">Britton Smith</a>)
**Repository:** <a href="https://github.com/ytree-project/ytree" target ="_blank">https://github.com/ytree-project/ytree</a>
**Version:** 2.3
**Editor:** @xuanxu
**Reviewer:** @mgckind, @aureliocarnero
**Archive:** <a href="https://doi.org/10.5281/zenodo.3580978" target="_blank">10.5281/zenodo.3580978</a>

## Status

[![status](https://joss.theoj.org/papers/5a26201bd9bf75e762f1f7b8d9ff198e/status.svg)](https://joss.theoj.org/papers/5a26201bd9bf75e762f1f7b8d9ff198e)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/5a26201bd9bf75e762f1f7b8d9ff198e"><img src="https://joss.theoj.org/papers/5a26201bd9bf75e762f1f7b8d9ff198e/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/5a26201bd9bf75e762f1f7b8d9ff198e/status.svg)](https://joss.theoj.org/papers/5a26201bd9bf75e762f1f7b8d9ff198e)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/ytree-project/ytree" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@mgckind & @aureliocarnero, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @xuanxu know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ytree-project/ytree">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@brittonsmith) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ytree-project/ytree">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@brittonsmith) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @mgckind, @aureliocarnero it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01881/joss.01881/10.21105.joss.01881.pdf)

**Comment by xuanxu** [comment]
@brittonsmith, @mgckind, @aureliocarnero: this is the review thread for the paper. All of our communications will happen here from now on.

Both reviewers have checklists at the top of this thread with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention openjournals/joss-reviews#1881 so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within about 2-4 weeks. Please let me know if any of you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me (@xuanxu) if you have any questions/concerns.

**Comment by aureliocarnero** [comment]
Hello @brittonsmith.
Your code does not contain a plain-text LICENSE file with the contents of an OSI approved software license.
Please add one to the repository in order to continue with the review.
Cheers

**Comment by brittonsmith** [comment]
@aureliocarnero, sure thing, done!

**Comment by aureliocarnero** [comment]
Thank you @brittonsmith - Could you please clarify me which license from this list, https://opensource.org/licenses/alphabetical correspond to yours.? Cheers

**Comment by aureliocarnero** [comment]
@xuanxu I cannot open the draft: 
https://github.com/openjournals/joss-papers/blob/joss.01881/joss.01881/10.21105.joss.01881.pdf

**Comment by aureliocarnero** [comment]
@brittonsmith I'm having troubles installing your code with pip.
It tells me the following:
  Stored in directory: /home/carnero/.cache/pip/wheels/63/9d/8e/37c3f6506ed3f152733a699e92d8e0c9f5e5f01dea262f80ad
Successfully built mpmath
Installing collected packages: mpmath, sympy, yt, ytree
  Found existing installation: sympy 0.7.6.1
ERROR: Cannot uninstall 'sympy'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.

Any idea how can I solve this?

**Comment by aureliocarnero** [comment]
@brittonsmith also, using the git clone, pip install instruction for the developer version, I also found this problem (maybe related)

ERROR: Exception:
Traceback (most recent call last):
  File "/home/carnero/anaconda2/lib/python2.7/site-packages/pip/_internal/cli/base_command.py", line 188, in main
    status = self.run(options, args)
  File "/home/carnero/anaconda2/lib/python2.7/site-packages/pip/_internal/commands/install.py", line 345, in run
    resolver.resolve(requirement_set)
  File "/home/carnero/anaconda2/lib/python2.7/site-packages/pip/_internal/legacy_resolve.py", line 196, in resolve
    self._resolve_one(requirement_set, req)
  File "/home/carnero/anaconda2/lib/python2.7/site-packages/pip/_internal/legacy_resolve.py", line 359, in _resolve_one
    abstract_dist = self._get_abstract_dist_for(req_to_install)
  File "/home/carnero/anaconda2/lib/python2.7/site-packages/pip/_internal/legacy_resolve.py", line 291, in _get_abstract_dist_for
    req, self.require_hashes, self.use_user_site, self.finder,
  File "/home/carnero/anaconda2/lib/python2.7/site-packages/pip/_internal/operations/prepare.py", line 255, in prepare_editable_requirement
    req.check_if_exists(use_user_site)
  File "/home/carnero/anaconda2/lib/python2.7/site-packages/pip/_internal/req/req_install.py", line 418, in check_if_exists
    self.req.name
  File "/home/carnero/anaconda2/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 481, in get_distribution
    dist = get_provider(dist)
  File "/home/carnero/anaconda2/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 357, in get_provider
    return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0]
  File "/home/carnero/anaconda2/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/carnero/anaconda2/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
ContextualVersionConflict: (yt 3.3.5 (/home/carnero/anaconda2/lib/python2.7/site-packages), Requirement.parse('yt>=3.4'), set(['ytree']))

**Comment by brittonsmith** [comment]
@aureliocarnero, the license is the [BSD-3-Clause](https://opensource.org/licenses/BSD-3-Clause).

For the installation issues, I'm not entirely sure. In the first instance, it looks like sympy was installed in a manner that cannot be upgraded by pip. In the second case, it looks like there is a conflicting version of yt that also cannot be updated. My recommendation would be to try creating a fresh conda environment from which to try installing from pip or source. I've just tried this now myself and it's working for me. Barring that, you might need to reinstall anaconda/miniconda. Please let me know if none of those suggestions work.

**Comment by aureliocarnero** [comment]
Hello @brittonsmith Im trying to download the data from here: https://ytree.readthedocs.io/en/latest/Data.html trying both the manual download and the one using the girder-cl app and it failed in both cases.
Could you please check that the data is correct? Maybe it is just me, but it would be good to know.
Cheers

**Comment by aureliocarnero** [comment]
Also, in terms of documentation, I think more information should be given in the landing page of the github. In my opinion, you do not explain well what are the application of the trees. May I suggest you explain the context is used, in terms of cosmological simulations. If it can be used for other applications than that, please explain as well

**Comment by aureliocarnero** [comment]
You should also explain better what this mean, please:

-    loading merger-tree data from the Amiga Halo Finder, Consistent-Trees, Rockstar, LHaloTree, and treefarm formats
-    on-demand loading of trees and fields
-    symbolic units, derived fields, and alias fields
-    saving trees to a universal format

Plus you have a typo, correct the following error:  see the full ytree documenation.

Cheers
Aurelio

**Comment by aureliocarnero** [comment]
In term of the paper, it would be good to give it a last review on the text to improve readability.

**Comment by aureliocarnero** [comment]
So @brittonsmith please read comments above.
Some changes in the documentation front needs to be done before I give green light for the publication.
Still I need to mark:
1) A statement of need: Do the authors clearly state what problems the software is designed to solve and who the target audience is?
Please improve the landing page introduction stating clearly what the program is for.
2) Quality of writing: Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
Please give a last reading to the paper, improving it a bit

**Comment by brittonsmith** [comment]
@aureliocarnero, sorry I was absolutely swamped last week and also wasn't sure if I should wait for a second review. Thank you for your comments. I'm happy to clarify the text on the landing page as per all of your suggestions there. I will take care of everything this week.

As to your second comment above, I'm not sure I understand what you would like to see improved about the paper itself. Could you please clarify this?

**Comment by xuanxu** [comment]
@mgckind: How is your review going?

**Comment by mgckind** [comment]
> @mgckind: How is your review going?

Hi @xuanxu , I got it installed and started looking at the documentation  (and getting some data bbut that went too slow which I'll revisit ) and reading @aureliocarnero comments. I've been traveling more than I planned to. I'll get this done as soon as I can, thanks!

**Comment by brittonsmith** [comment]
@aureliocarnero, I've issued a [pull request](https://github.com/ytree-project/ytree/pull/34) to address your comments on the landing page. I chose to remove the bullet point list of features and to replace it with a short code demo as that seemed more illustrative.

Can you provide some more specific comments on how you'd like to see the paper improved?

**Comment by aureliocarnero** [comment]
I answered through github I think. The README is much better, thanks. I
will send you comments about the paper shortly.

**Comment by aureliocarnero** [comment]
@brittonsmith about improving the readability of the paper. I think it could be improve, for example, I suggest the following changes:

In the phrase:
1) In the early Universe, matter is very evenly distributed, with only small fluxuations
about the average density. 
Change to 
In the early Universe, matter is distributed homogeneously, with only small fluctuations
about the average density.

2) Halos grow via accretion of the surrounding material and mergers with other halos.
Change to
Halos grow via accretion of the surrounding material and by merging with other halos.

3) This process of merging to form increasingly massive halos is very naturally conceptualized as an inverted tree, where small branches connect up to continually larger ones, leading eventually to the trunk.
Change to
This process of merging to form increasingly massive halos is naturally conceptualized as an inverted tree, where small branches connect up to continually larger ones, leading eventually to a trunk.

4) One of the main products of a cosmological simulation is a series of catalogs of all halos within
the simulated volume at a number of epochs.
Change to
One of the main products of cosmological simulations is a series of catalogs of halos within
the simulated volume at different epochs.

5) Halos within succesive catalogs can be linked together to create merger trees that describe a halo’s growth history.
Change to
Halos within successive epochs can be linked together to create merger trees that describe a halo’s growth history.

6) About figure 1. It is confusing for, according to the text, first small halos are formed and then more massive halos results. According to the caption, it seems the contrary. According to the explanation in the text, descendents are going upward, not downward. Could you check this please? Also, it would be nice if on the left of the plot, you put an arrow representing time, this would facilitate the explanation

Cheers

**Comment by brittonsmith** [comment]
@aureliocarnero, thanks for your suggestions on the paper. I've implemented them all, including for Figure 1, in [this PR](https://github.com/ytree-project/ytree/pull/35). The wording in the Figure caption was a mistake; I had ancestor and descendents switched. It should make sense now. Anyway, please let me know if that looks ok.

**Comment by brittonsmith** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01881/joss.01881/10.21105.joss.01881.pdf)

**Comment by aureliocarnero** [comment]
After checking all the issues, I consider this publication is ready for submission and publication to JOSS

**Comment by mgckind** [comment]
I've started with the review. I should be done by today/tomorrow.

**Comment by mgckind** [comment]
@brittonsmith , @xuanxu.  I'm sorry again for being late but here I am.
Great package and nice documentation, I went through every point, I checked aurelio's comments and installed and used `ytree`. Everything looks good to me. 
 I've created 2 issues in the original repo in regards to the [paper](https://github.com/ytree-project/ytree/issues/36) and some [functionalities](https://github.com/ytree-project/ytree/issues/37). These are minor suggestions, other than that and after @aureliocarnero comments being addressed the paper looks good.  Congrats!

**Comment by brittonsmith** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01881/joss.01881/10.21105.joss.01881.pdf)

**Comment by brittonsmith** [comment]
@mgckind, I've just issued a new [pull request](https://github.com/ytree-project/ytree/pull/39) to address your [functionality comments](https://github.com/ytree-project/ytree/issues/37). Please let me know if that looks ok.

**Comment by mgckind** [comment]
I went through the PR and it looks awesome, thanks for the extra effort of adding those. Congrats on such a nice module!. @xuanxu , @brittonsmith This is ready to go.

**Comment by brittonsmith** [comment]
@mgckind, thanks so much and thank you for reviewing the latest PRs so quickly!

**Comment by xuanxu** [comment]
Thanks @aureliocarnero and @mgckind for your review!

**Comment by xuanxu** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01881/joss.01881/10.21105.joss.01881.pdf)

**Comment by xuanxu** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1088/0067-0049/192/1/9 is OK
- 10.5281/zenodo.1174374 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.21105/joss.00809 is OK
- 10.1002/1097-024X(200009)30:11<1203::AID-SPE338>3.0.CO;2-N is OK
- 10.1088/0004-637X/763/1/18 is OK
- 10.3847/1538-4357/aabe8f is OK
- 10.3847/1538-4357/ab378f is OK
- 10.1093/mnras/sty2103 is OK
- 10.1093/mnras/stz2507 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by xuanxu** [comment]
OK @brittonsmith, everything looks good, here are the next steps:

- Please release a new tagged version from the current master so it includes all the changes made during the review process
- Then archive that latest release in Zenodo
- Check the Zenodo deposit has the correct metadata: title and author name should match the paper; you may also add your ORCID.

Once you do that please report here the version number and archive DOI

**Comment by brittonsmith** [comment]
Hi @xuanxu, I've just done the release.

Version: 2.3
DOI: [10.5281/zenodo.3580978](https://doi.org/10.5281/zenodo.3580978)

The metadata for the Zenodo entry looks good to me. Our ORCIDs are there, too. I think we're ready to go.

Thanks for all your help!

**Comment by xuanxu** [comment]
@whedon set 2.3 as version

**Comment by whedon** [comment]
OK. 2.3 is the version.

**Comment by xuanxu** [comment]
@whedon set 10.5281/zenodo.3580978 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3580978" target="_blank">10.5281/zenodo.3580978</a> is the archive.

**Comment by xuanxu** [comment]
Thanks @brittonsmith!
Everything is ready :tada: 
@openjournals/joss-eics sending it your way for final acceptance! :rocket:

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1088/0067-0049/192/1/9 is OK
- 10.5281/zenodo.1174374 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.21105/joss.00809 is OK
- 10.1002/1097-024X(200009)30:11<1203::AID-SPE338>3.0.CO;2-N is OK
- 10.1088/0004-637X/763/1/18 is OK
- 10.3847/1538-4357/aabe8f is OK
- 10.3847/1538-4357/ab378f is OK
- 10.1093/mnras/sty2103 is OK
- 10.1093/mnras/stz2507 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1178

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1178, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1207305966783533056) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1179
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01881](https://doi.org/10.21105/joss.01881)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by arfon** [comment]
@mgckind, @aureliocarnero - many thanks for your reviews here and to @xuanxu for editing this submission ✨ 

@brittonsmith - your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01881/status.svg)](https://doi.org/10.21105/joss.01881)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01881">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01881/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01881/status.svg
   :target: https://doi.org/10.21105/joss.01881
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01881/status.svg)](https://doi.org/10.21105/joss.01881)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by brittonsmith** [comment]
Thanks everyone for all their work! This was a great experience.
